source/agora/test/BanManager.d

```diff
@@ -69,71 +70,41 @@ unittest
     node_1.filter!(node_1.getBlocksFrom);  // node 2 can't retrieve blocks
     node_2.filter!(node_2.putTransaction); // node 1 can't gossip transactions
 
-    // node 2 will reject gossipping
+    // leftover txs which node 2 rejected due to filter
     Transaction[] left_txs;
 
-    auto new_tx = genBlockTransactions(1);
+    auto new_tx = genBlockTransactions(4);
     left_txs ~= new_tx;
     new_tx.each!(tx => node_1.putTransaction(tx));
-    Thread.sleep(500.msecs);
 
-    new_tx = genBlockTransactions(1);
-    left_txs ~= new_tx;
-    new_tx.each!(tx => node_1.putTransaction(tx));
-    Thread.sleep(500.msecs);
-
-    new_tx = genBlockTransactions(1);
-    left_txs ~= new_tx;
-    new_tx.each!(tx => node_1.putTransaction(tx));
-    Thread.sleep(500.msecs);
-
-    new_tx = genBlockTransactions(1);
-    left_txs ~= new_tx;
-    new_tx.each!(tx => node_1.putTransaction(tx));
-    Thread.sleep(500.msecs);
-    assert(node_1.getBlockHeight() == 5);
-    assert(node_2.getBlockHeight() == 1);
-
-    node_2.clearFilter();
-    left_txs.each!(tx => node_2.putTransaction(tx));  // manually add them
-    Thread.sleep(500.msecs);
-    assert(node_1.getBlockHeight() == 5);
-    assert(node_2.getBlockHeight() == 5);
-
-    new_tx = genBlockTransactions(1);
-    left_txs ~= new_tx;
-    new_tx.each!(tx => node_1.putTransaction(tx));
-    Thread.sleep(500.msecs);
-    assert(node_1.getBlockHeight() == 6);
-    assert(node_2.getBlockHeight() == 5);  // filter is active
+    // wait for node 2 to be banned and all putTransaction requests to time-out
+    Thread.sleep(2.seconds);
```

Comment:
The timeout is `10` (msecs?) so why wait 2 seconds ?

Reply:
It keeps polling with a 10 msec timeout until a total of 2 seconds are reached. It will either finish early or throw.

Reply:
I might have forgotten the details here, so my bad, but what will lead to node 2 being banned ?

Reply:
32 failed putTransaction's.